RIPS/rip-9999.md

```diff
@@ -0,0 +1,317 @@
+This means that the "implementation" contract is the entity responsible for implementing all kinds of protections,
+and the EOA account that signs an authorization fully cedes control to the "implementation" until
+the authorization is revoked.
+
+For the purposes of Gas Abstraction, the `paymasterData` parameter is not signed as well,
+so different Paymaster contracts can be used for subsequent authorized transactions using the same authorization.
+
+#### Using permanent authorizations to create Smart Contract Accounts
+
+This approach relies on the so-called "Nick's method", which includes deriving an address from some data and
+a randomized signature value, without ever knowing the ECDSA private key that would be able to control this address.
+
+Applying this approach to an `AUTHORIZED_EOA_SUBTYPE` transaction, a new account can be created,
+whose address is forever bound to `implementationAddress` and `commit`,
+without ever needing to deploy code for the contract.
+
+### Gas-sponsored time-limited code injection EOA transaction subtype
+
+```
+INSERT_CODE_EOA_SUBTYPE
+```
+
+This transaction subtype extends the functionality of RIP-7560 EOA transactions with an ability to insert
+contract code in the context of the EOA sender.
+
+The transaction payload for this subtype is extended with the following fields:
+```
+callGasLimit,
+initCode,
+value,
+yParity, r, s
+```
+
+The `initCode` is executed in the context of the account address that is recovered from the
+signature parameters (`yParity`, `r`, `s`).
+
+Same as the regular transaction, the data returned by executing the `initCode` is stored as the contract code
+for the signer's account address.
+
+In case the returned value is not a valid contract code the transaction reverts all changes except for gas charge.
+
+In case the signer's account address contains code before `initCode` runs the transaction reverts immediately.
+
+## Rationale
+
+### Authorization as a transaction subtype instead of a family of opcodes
+
+Both approaches have their benefits and downsides.
+Here are some of the benefits of a transaction subtype approach:
+
+* Use of `AUTH` and `AUTHCALL` opcodes only allow an unnecessarily dangerous way for EOAs to achieve Gas Abstraction.
+
+    While technically it is possible to implement a gas relaying feature on top of EIP-3074, this approach requires
+users to fully hand over control of their accounts to the gas paying "invoker" contract.
+A lot can go wrong with this approach.
+
+    Another benefit is that it allows smart accounts and EOAs to use the same paymasters.
+
+    A dapp that sponsors a transaction doesn't have to care whether the user has a smart account or an EOA.
+    And paymasters like TokenPaymaster could allow both smart accounts and EOAs to pay with
+    [ERC-20](./eip-20) tokens.
+
+    The 3074-based approach, in contrast, would require developing separate gas abstraction invokers and dapps
+    would have to be aware of the user's account type.
```

Comment:
I don't think this is true. 4337 can be extended to have an invoker which performs the EOA subtype transactions described in this document. If 4337 decides not to change, then this may be true. In which case it should be noted.